src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/ShipOutOfLuck.java

```diff
@@ -82,56 +77,44 @@ public void preventBoatPlacements(PlayerInteractEvent event) {
 	}
 
 	@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
-	public void preventBoatUsage(VehicleMoveEvent event) {
-		Vehicle vehicle = event.getVehicle();
+	public void preventBoatUsage(final VehicleMoveEvent event) {
+		final Vehicle vehicle = event.getVehicle();
 		if (vehicle.getType() != EntityType.BOAT) {
 			return;
 		}
-		List<Entity> passengers = vehicle.getPassengers();
+		final List<Entity> passengers = vehicle.getPassengers();
 		if (passengers.isEmpty()) {
 			return;
 		}
-		Location destination = event.getTo();
-		if (event.getVehicle().isOnGround()) {
+		final Location destination = event.getTo();
+		if (vehicle.isOnGround()) {
 			// If the boat is on top of a block, then we need to look below the boat
 			destination.add(0, -1, 0);
 		}
 		if (destination.getY() < 0 || destination.getY() >= destination.getWorld().getMaxHeight()) {
 			return;
 		}
-		Block currentBlock = destination.getBlock();
-		if (!this.config.isBoatBreaker(currentBlock.getType())) {
-			return;
+		final Block currentBlock = destination.getBlock();
+		final Material currentMaterial = currentBlock.getType();
+		if (this.boatBreakers.contains(currentMaterial)) {
+			vehicle.eject();
+			plugin().debug("Ejected [" + passengers.stream().map(CommandSender::getName)
+					.collect(Collectors.joining(", ")) + "] from boat at [" + destination + "] because " +
+					"they sailed over [" + currentMaterial.name() + "]");
+		}
+		else if (!this.incongruousBoats
+				&& vehicle.isOnGround()
+				&& MaterialAPI.isWithoutSubstance(currentMaterial)) {
```

Comment:
Not a very elegant solution imo, it's very common to have air below your boat when doing any kind of downwards y-traversal (which isn't too uncommon). I'd rather scan for boatBreaker blocks in a cross shape around the location.

Reply:
This doesn't just check for air, it checks for the incongruous situation where the boat considers itself on the ground while being above air, or an air-like block.

Reply:
![Proof](https://user-images.githubusercontent.com/52572989/99587802-8aeb7800-29e1-11eb-9da1-954a9ae6a824.png)

Reply:
Oh okay I see, nevermind then